### PR TITLE
AIEye Cleanup

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
This PR moves the AIEye from being /mob/dead to /mob/living/intangible, and makes (hopefully) all the necessary changes to preserve behaviour.

## Why's this needed? 
Having the AI eye register as dead seems fundamentally wrong, given that an intangible living class exists. This should make future code easier (because you no longer have to have exceptions for interacting with a single dead mob) and also make it possible to remove and generally clean up checks in a lot of places. 
This also fixes some minor issues with the blob overmind, such as preventing the mining magnet from working - the safety lock detects /mob/living but did not make an exception for intangible.

Generalising classes = good, makes code more maintainable - less edge cases.

EDIT (no changelog):

Bugs also fixed by this PR:
(+)Fixed the blob overmind being able to pull objects
(+)Fixed the blob overmind being able to block the mining magnet by sitting in the mining area
(+)Replaced a lot of istype(AIEye) calls with the appropriate macro
(+)Fixed a bug where the blob overmind would be repeatedly tripped by the syndicate chair

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Okay but does this fix being able to drag the AI eye if you can see ghosts, because I love that "feature"

**Comment by ThatFiveGuys** [comment]
Does this also “fix” ai being able to rotate chairs and interact with the ouija board? Changelog seems way too long too, generally bug fixes dont need a changelog.

**Comment by amylizzle** [comment]
> Okay but does this fix being able to drag the AI eye if you can see ghosts, because I love that "feature"

I was not aware that was a thing, and so probably not. /atom/movable's pull only checks if the puller is intangible, not if the thing being pulled is - though it probably should.



> Does this also “fix” ai being able to rotate chairs and interact with the ouija board? Changelog seems way too long too, generally bug fixes dont need a changelog.

Ouija board was explicitly re-added on request of Pali. Chairs should still be rotatable.

File: code/obj/mining.dm [hunk 1]
```diff
@@ -779,7 +779,7 @@
 			return 1
 
 		var/rangecheck = 0
-		if (issilicon(usr))
+		if (isAI(usr))
```
Comment (ZeWaka) [inline code/obj/mining.dm:782]: This stops borgs from using it

File: code/modules/interface/input.dm [hunk 2]
```diff
@@ -175,7 +175,7 @@ var/list/dirty_keystates = list()
 
 		if(findtext( control, "viewport" ))
 			var/datum/viewport/vp = getViewportById(control)
-			if(vp && vp.clickToMove && object && isturf(object) && (mob.type == /mob/living/intangible/blob_overmind || mob.type == /mob/dead/aieye))//NYI: Replace the typechecks with something Better.
+			if(vp && vp.clickToMove && object && isturf(object) && (mob.type == /mob/living/intangible/blob_overmind || mob.type == /mob/living/intangible/aieye))//NYI: Replace the typechecks with something Better.
```
Comment (ZeWaka) [inline code/modules/interface/input.dm:178]: istype() preferred

**Comment by ZeWaka** [comment]
also generally bug fixes or non-player-facing changes don't go in the changelog re: contributor guidelines

File: code/obj/mining.dm (hunk 1 above)
Comment (amylizzle) [reply code/obj/mining.dm:782]: isAI() checks for silicon or AIeye
`#define isAI(x) (istype(x, /mob/living/silicon/ai) || istype (x, /mob/living/intangible/aieye))`

File: code/modules/interface/input.dm (hunk 2 above)
Comment (amylizzle) [reply code/modules/interface/input.dm:178]: should probably just replace` (mob.type == /mob/living/intangible/blob_overmind || mob.type == /mob/living/intangible/aieye))`
with `isintangible(mob)`

**Review by pali6 — APPROVED** [review]
At a glance looks good. I'm sure this will cause weird edge case bugs at some point down the line but we'll fix those as they appear.
Also it most likely fixes weird edge case bugs too hehe.

File: code/obj/mining.dm (hunk 1 above)
Comment (pali6) [reply code/obj/mining.dm:781]: It checks for silicon/**ai** not silicon.

File: code/obj/mining.dm [hunk 3]
```diff
@@ -778,7 +778,7 @@
...
```
Comment (pali6) [inline code/obj/mining.dm:781]:
```suggestion
		if (isAI(usr) || issilicon(usr))
```

**Pull request merged**